src/classes/CallableApi.cls

```diff
@@ -0,0 +1,144 @@
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+ 
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group API
+* @group-content ../../ApexDocContent/API.htm
+* @description API class that exposes a callable class for users
+*/
+global class CallableApi implements System.Callable {
+
+    /*******************************************************************************************************
+    * @description Exception class for Malformed Method Invocations
+    ********************************************************************************************************/
+    global class MalformedMethodInvocationException extends Exception {}
+
+    /*******************************************************************************************************
+    * @description Map of valid actions used to action text
+    ********************************************************************************************************/
+    private static Map<String, String> validActionsMap = new Map<String, String> {
+        'tdtm.disablealltriggers' => 'TDTM.disableAllTriggers'
+    };
+
+    /*******************************************************************************************************
+    * @description call function implementation of the callable Interface will dispatch to the appropriate
+    * action handler based on the action text
+    * @param action string of the target action
+    * @param args name value map of arguments
+    * @return Object returned will depend on the results of the target function
+    ********************************************************************************************************/
+    global Object call(String action, Map<String, Object> args) {
+       
+        CallableApiInfo actionInfo = validateAction(action, args);
+        
+        if (actionInfo == null) {
```

Comment:
I'm not sure there's a lot of value gained by validating the action and then calling another method with the switch statement. I think you can collapse/simplify all of that by just using a single switch statement in this method. The `when else` part would cover an invalid action. With that model you can eliminate the map (which also eliminates duplicating the action names), the wrapper class and the other two methods. I think it would be a lot easier to read as well.